lib/matplotlib/tests/test_legend.py

```diff
@@ -1375,3 +1377,16 @@ def test_legend_handle_label_mismatch_no_len():
                         labels=iter(["pl1", "pl2", "pl3"]))
     assert len(legend.legend_handles) == 2
     assert len(legend.get_texts()) == 2
+
+
+def test_legend_nolabels_warning():
+    plt.plot([1, 2, 3])
+    with pytest.raises(UserWarning, match="No artists with labels found"):
+        plt.legend()
+
+
+def test_legend_nolabels_draw():
+    plt.plot([1, 2, 3])
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        plt.legend()
```

Comment:
Let's at least make sure, a legend is created and attached to the Axes:
```suggestion
        plt.legend()
    assert plt.gca().get_legend() is not None
```

Reply:
Gotcha, makes sense!  In addition, is pattern you suggested above of `@pytest.mark.filterwarnings(...)` preferable to `warnings.catch_warnings()`+`simplefilter` pattern?